test/apicoverage-webhook/webhook_handlers.go

```diff
@@ -0,0 +1,173 @@
+		err := json.Unmarshal(review.Request.Object.Raw, &svc)
+		if err != nil {
+			a.Logger.Errorf("Error unmarshalling Service Obj. Request Obj: %s Error: %v", review.Request.Object.Raw, err)
+			return
+		}
+		resourceTree := a.resourceForest.TopLevelTrees["Service"]
+		resourceTree.UpdateCoverage(reflect.ValueOf(svc))
+	case "Configuration":
+		conf := v1alpha1.Configuration{}
+		err := json.Unmarshal(review.Request.Object.Raw, &conf)
+		if err != nil {
+			a.Logger.Errorf("Error unmarshalling Service Obj. Request Obj: %s Error: %v", review.Request.Object.Raw, err)
+			return
+		}
+		resourceTree := a.resourceForest.TopLevelTrees["Configuration"]
+		resourceTree.UpdateCoverage(reflect.ValueOf(conf))
+	case "Route":
+		route := v1alpha1.Route{}
+		err := json.Unmarshal(review.Request.Object.Raw, &route)
+		if err != nil {
+			a.Logger.Errorf("Error unmarshalling Service Obj. Request Obj: %s Error: %v", review.Request.Object.Raw, err)
+			return
+		}
+		resourceTree := a.resourceForest.TopLevelTrees["Route"]
+		resourceTree.UpdateCoverage(reflect.ValueOf(route))
+	case "Revision":
+		revision := v1alpha1.Revision{}
+		err := json.Unmarshal(review.Request.Object.Raw, &revision)
+		if err != nil {
+			a.Logger.Errorf("Error unmarshalling Service Obj. Request Obj: %s Error: %v", review.Request.Object.Raw, err)
+			return
+		}
+		resourceTree := a.resourceForest.TopLevelTrees["Revision"]
+		resourceTree.UpdateCoverage(reflect.ValueOf(revision))
+	}
+}
+
+// Since we are not actually validating any API resource object we always return true here.
+func (a *APICoverageRecorder) appendReviewResponse(review *v1beta1.AdmissionReview) {
+	review.Response = &v1beta1.AdmissionResponse{
+		Allowed: true,
+		Result: &v1.Status{
+			Message: "Welcome aboard!",
+		},
+	}
+}
+
+// RecordResourceCoverage is handler method that records resource coverage.
+func (a *APICoverageRecorder) RecordResourceCoverage(w http.ResponseWriter, r *http.Request) {
+	var body []byte
+	if data, err := ioutil.ReadAll(r.Body); err == nil {
+		body = data
+	}
+
+	review := &v1beta1.AdmissionReview{}
+	_, _, err := a.Decoder.Decode(body, nil, review)
+	if err != nil {
+		a.Logger.Errorf("Can't decode request : %v", err)
+	}
+	a.recordCoverage(review)
+	a.appendReviewResponse(review)
+	responseInBytes, err := json.Marshal(review)
+
+	if _, err := w.Write(responseInBytes); err != nil {
+		a.Logger.Errorf("%v", err)
+	}
+}
+
+// GetResourceCoverage retrieves resource coverage data for the passed in resource via query param.
+func (a *APICoverageRecorder) GetResourceCoverage(w http.ResponseWriter, r *http.Request) {
+	resource := r.URL.Query().Get(ResourceQueryParam)
+	if _, ok := a.resourceForest.TopLevelTrees[resource]; !ok {
+		fmt.Fprintf(w, "Resource information not found for resource : %s", resource)
+		return
+	}
+
+	var ignoreFields coveragecalculator.IgnoredFields
+	ignoreFieldsFilePath := "resources/ignoredfields.yaml"
+	err := ignoreFields.ReadFromFile(ignoreFieldsFilePath)
```

Comment:
You can probably join these two lines

Reply:
The path is used in the log statement below, hence the variable.

Reply:
Ahh I see,  sorry thought it was for the lines before. updated it.